File: src/types/document.ts
// Document types - aligned with ARCHITECTURE.md database schema
export interface Document {
  id: number;
  title: string;
  content: string;
  version: number;
  tags?: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateDocument {
  title: string;
  content: string;
  tags?: string | null;
}

export interface DocumentVersion {
  id: number;
  document_id: number;
  content: string;
  version: number;
  created_at: string;
}

File: src-tauri/src/models/document.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Document {
    pub id: i64,
    pub title: String,
    pub content: String,
    pub version: i64,
    pub tags: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateDocument {
    pub title: String,
    pub content: String,
    pub tags: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateDocument {
    pub title: Option<String>,
    pub content: Option<String>,
    pub tags: Option<String>,
}

